hyperspy/hspy.py

```diff
@@ -97,11 +97,9 @@ def create_model(signal, *args, **kwargs):
     A Model class
 
     """
-
-    from hyperspy._signals.eels import EELSSpectrum
-    from hyperspy.models.eelsmodel import EELSModel
-    from hyperspy.model import Model
-    if isinstance(signal, EELSSpectrum):
-        return EELSModel(signal, *args, **kwargs)
-    else:
-        return Model(signal, *args, **kwargs)
+    import warnings
+    warnings.warn(
+        "This function is deprecated and will be removed in HyperSpy 0.9. "
+        "Please use the equivalent `Signal.create_model` method "
+        "instead.", DeprecationWarning)
```

Comment:
Yes, I think that the issue is more general, so I agree that it'll be
better to let this in and fix the general issue in a different PR.
An alternative would be to use numpy.VisibleDeprecationWarning.
On Mon, 2015-07-27 at 07:24 -0700, vidartf wrote:
> In hyperspy/hspy.py:
> > @@ -97,11 +97,9 @@ def create_model(signal, *args, **kwargs):
> >      A Model class
> >  
> >      """
> > -
> > -    from hyperspy._signals.eels import EELSSpectrum
> > -    from hyperspy.models.eelsmodel import EELSModel
> > -    from hyperspy.model import Model
> > -    if isinstance(signal, EELSSpectrum):
> > -        return EELSModel(signal, *args, **kwargs)
> > -    else:
> > -        return Model(signal, *args, **kwargs)
> > +    import warnings
> > +    warnings.warn(
> > +        "This function is deprecated and will be removed in
> HyperSpy 0.9. "
> > +        "Please use the equivalent `Signal.create_model` method "
> > +        "instead.", DeprecationWarning)
> I think I'll make this a separate issue, and merge this. Agreed?
> —
> Reply to this email directly or view it on GitHub.